include/pybind11/attr.h

```diff
@@ -14,27 +14,22 @@
 
 NAMESPACE_BEGIN(pybind11)
 
-template <typename T> struct arg_t;
-
 /// Annotation for keyword arguments
 struct arg {
     constexpr explicit arg(const char *name) : name(name) { }
 
     template <typename T>
     constexpr arg_t<T> operator=(const T &value) const { return {name, value}; }
-    template <typename T, size_t N>
-    constexpr arg_t<const T *> operator=(T const (&value)[N]) const {
-        return operator=((const T *) value);
-    }
 
     const char *name;
 };
 
 /// Annotation for keyword arguments with default values
 template <typename T> struct arg_t : public arg {
     constexpr arg_t(const char *name, const T &value, const char *descr = nullptr)
-        : arg(name), value(value), descr(descr) { }
-    T value;
+        : arg(name), value(&value), descr(descr) { }
```

Comment:
Hmm, this change (pointer instead of an explicit copy of ``value``) seems problematic to me. It means that ``arg_t`` cannot receive something that is constructed on the fly, e.g.

```
m.def("fun", py::arg("arg") = MyArg(1, 2, 3));
```